src/components/common/NetworkSelector/index.tsx

```diff
@@ -66,25 +66,32 @@ export const getNetworkLink = (router: NextRouter, networkShortName: string, isW
 }
 
 const UndeployedNetworkMenuItem = ({
-  chainId,
-  chainConfigs,
+  chain,
   isSelected = false,
   onSelect,
 }: {
-  chainId: string
-  chainConfigs: ChainInfo[]
+  chain: ChainInfo & { available: boolean }
   isSelected?: boolean
   onSelect: (chain: ChainInfo) => void
 }) => {
-  const chain = useMemo(() => chainConfigs.find((chain) => chain.chainId === chainId), [chainConfigs, chainId])
-
-  if (!chain) return null
```

Comment:
Not sure why we needed to pass `chains` here just to filter for the current chain again since we iterate over `prodNets` and `testNets` we have access to the chain anyway.